### ISSUE TITLE
interactive variables not updating due to no default values in list.assoc

### ISSUE THREAD

**Issue by Russsgithub** [savonet/liquidsoap#4194]
### Description

Using interactive.bool results in 

```
Could not update variable  (not_found: no default value for list.assoc).


(no other information is given in the error to determine where this is occuring, but i have no calls to list.assoc in my script.)
```

When i update the variable through the :8000/interactive interface.

I believe this also occurs with the other interactive operators.

### Steps to reproduce

s =sine(440.)

bool = interactive.bool("toggle", false)

output.dummy(fallible=true, s)

### Expected behavior

bool respond to external call to change.

### Liquidsoap version

```text
Liquidsoap rolling-release-v2.3.x-54-g63aea1b+dev
Copyright (c) 2003-2024 Savonet team
Liquidsoap is open-source software, released under GNU General Public License.
See <http://liquidsoap.info> for more information.
```

### Liquidsoap build config

```text
* Liquidsoap version  : rolling-release-v2.3.x-54-g63aea1b+dev

 * Compilation options
   - Release build       : false
   - Git SHA             : (none)
   - OCaml version       : 4.14.2
   - OS type             : Unix
   - Libs versions       : angstrom=0.16.1 asetmap=0.8.1 asn1-combinators=0.3.2 astring=0.8.5 base=v0.16.3 base.base_internalhash_types=v0.16.3 base.caml=v0.16.3 base.shadow_stdlib=v0.16.3 base64=
3.5.1 bigarray=[distributed with Ocaml] bigarray-compat=1.1.0 bigstringaf=0.10.0 bos=0.2.1 bytes=[distributed with OCaml 4.02 or above] ca-certs=v1.0.0 camlp-streams camomile.lib=2.0 cohttp=5.3.1 
cohttp-lwt=5.3.0 cohttp-lwt-unix=5.3.0 conduit=7.1.0 conduit-lwt=7.1.0 conduit-lwt-unix=7.1.0 cry=1.0.3 cstruct=6.2.0 ctypes=0.23.0 ctypes-foreign=0.23.0 curl=0.9.2 digestif.c=1.2.0 domain-name=0.
4.0 domain_shims dtools=0.4.5 dune-build-info=3.16.0 dune-private-libs.dune-section=3.16.0 dune-site=3.16.0 dune-site.private=3.16.0 duppy=0.9.4 eqaf=0.10 ffmpeg-av=1.2.0 ffmpeg-avcodec=1.2.0 ffmp
eg-avdevice=1.2.0 ffmpeg-avfilter=1.2.0 ffmpeg-avutil=1.2.0 ffmpeg-swresample=1.2.0 ffmpeg-swscale=1.2.0 fileutils=0.6.4 fmt=0.9.0 fpath=0.7.3 gen=1.1 gmap=0.3.0 integers ipaddr=5.6.0 ipaddr-sexp=
5.6.0 ipaddr.unix=5.6.0 kdf.hkdf=1.0.0 kdf.pbkdf=1.0.0 ladspa=0.2.2 liquidsoap-lang=rolling-release-v2.3.x-54-g63aea1b liquidsoap-lang.console=rolling-release-v2.3.x-54-g63aea1b liquidsoap_builtin
s=rolling-release-v2.3.x-54-g63aea1b-dirty liquidsoap_core=rolling-release-v2.3.x-54-g63aea1b-dirty liquidsoap_ffmpeg=rolling-release-v2.3.x-54-g63aea1b-dirty liquidsoap_ladspa=rolling-release-v2.
3.x-54-g63aea1b-dirty liquidsoap_optionals=rolling-release-v2.3.x-54-g63aea1b-dirty liquidsoap_oss=rolling-release-v2.3.x-54-g63aea1b-dirty liquidsoap_prometheus=rolling-release-v2.3.x-54-g63aea1b
-dirty liquidsoap_runtime=rolling-release-v2.3.x-54-g63aea1b-dirty liquidsoap_sdl=rolling-release-v2.3.x-54-g63aea1b-dirty liquidsoap_sdl_log_level=rolling-release-v2.3.x-54-g63aea1b-dirty liquids
oap_soundtouch=rolling-release-v2.3.x-54-g63aea1b-dirty liquidsoap_ssl=rolling-release-v2.3.x-54-g63aea1b-dirty liquidsoap_stdlib=rolling-release-v2.3.x-54-g63aea1b-dirty liquidsoap_stereotool=rol
ling-release-v2.3.x-54-g63aea1b-dirty liquidsoap_tls=rolling-release-v2.3.x-54-g63aea1b-dirty logs=0.7.0 logs.fmt=0.7.0 logs.lwt=0.7.0 lwt lwt.unix macaddr=5.6.0 magic-mime=1.3.1 mem_usage=0.1.1 m
enhirLib=20240715 metadata=0.3.0 mirage-crypto=1.1.0 mirage-crypto-ec=1.1.0 mirage-crypto-pk=1.1.0 mirage-crypto-rng=1.1.0 mirage-crypto-rng.unix=1.1.0 mm=0.8.5 mm.audio=0.8.5 mm.base=0.8.5 mm.ima
ge=0.8.5 mm.midi=0.8.5 mm.video=0.8.5 ocplib-endian ocplib-endian.bigstring ohex ppx_compare.runtime-lib=v0.16.0 ppx_hash.runtime-lib=v0.16.0 ppx_sexp_conv.runtime-lib=v0.16.0 prometheus=1.2 prome
theus-app=1.2 ptime=1.2.0 ptime.clock=1.2.0 ptime.clock.os=1.2.0 re=1.12.0 rresult=0.7.0 saturn_lockfree=0.4.1 sedlex=3.2 seq=[distributed with OCaml 4.07 or above] sexplib0=v0.16.0 soundtouch=0.1
.9 ssl=0.7.0 stdlib-shims=0.3.0 stereotool=rolling-release-v2.3.x-54-g63aea1b-dirty str=[distributed with Ocaml] stringext=1.6.0 threads=[distributed with Ocaml] threads.posix=[internal] tls=1.0.2
 tsdl=v1.1.0 tsdl-image=0.5 tsdl-ttf=0.6 unix=[distributed with Ocaml] uri=4.4.0 uri-sexp=4.4.0 uri.services=4.4.0 x509=1.0.4 zarith=1.14
   - architecture        : amd64
   - host                : x86_64-pc-linux-gnu
   - target              : x86_64-pc-linux-gnu
   - system              : linux
   - ocamlopt_cflags     : -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC
   - native_c_compiler   : gcc -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -D_FILE_OFFSET_BITS=64
   - native_c_libraries  : -lm

 * Configured paths
   - mode              : default
   - standard library  : (set by dune-site)
   - scripted binaries : (set by dune-site)
   - rundir            : (set by dune-site)
   - logdir            : (set by dune-site)
   - user cache        : $HOME/.cache/liquidsoap (override with $LIQ_CACHE_USER_DIR)
   - system cache      : (set by dune-site) (override with $LIQ_CACHE_SYSTEM_DIR)
   - camomile files    : (set by dune-site)

 * Supported input formats
   - MP3               : no (requires mad)
   - AAC               : no (requires faad)
   - Ffmpeg            : yes
   - Flac (native)     : no (requires flac)
   - Flac (ogg)        : no (requires ogg)
   - Opus              : no (requires opus)
   - Speex             : no (requires speex)
   - Theora            : no (requires theora)
   - Vorbis            : no (requires vorbis)
   - WAV/AIFF          : yes (native)

* Supported output formats
   - FDK-AAC           : no (requires fdkaac)
   - FFmpeg            : yes
   - MP3               : no (requires lame)
   - MP3 (fixed-point) : no (requires shine)
   - Flac (native)     : no (requires flac)
   - Flac (ogg)        : no (requires ogg)
   - Opus              : no (requires opus)
   - Speex             : no (requires speex)
   - Theora            : no (requires theora)
   - Vorbis            : no (requires vorbis)
   - WAV/AIFF          : yes (native)

 * Tags
   - AAC               : no (requires faad)
   - FFmpeg            : yes
   - FLAC (native)     : no (requires flac)
   - Flac (ogg)        : no (requires ogg)
   - Native decoder    : yes
   - Vorbis            : no (requires vorbis)

 * Input / output
   - ALSA              : no (requires alsa)
   - AO                : no (requires ao)
   - FFmpeg            : yes
   - JACK              : no (requires bjack)
   - OSS               : yes
   - Portaudio         : no (requires portaudio)
   - Pulseaudio        : no (requires pulseaudio)
   - SRT               : no (requires srt)

 * Audio manipulation
   - FFmpeg            : yes
   - LADSPA            : yes
   - Lilv              : no (requires lilv)
   - Samplerate        : no (requires samplerate)
   - SoundTouch        : yes
   - StereoTool        : yes

 * Video manipulation
   - camlimages        : no (requires camlimages)
   - FFmpeg            : yes
   - frei0r            : no (requires frei0r)
   - ImageLib          : no (requires imagelib)
   - SDL               : yes

 * MIDI manipulation
   - DSSI              : no (requires dssi)

 * Visualization
   - GD                : no (requires gd)
   - Graphics          : no (requires graphics)
   - SDL               : yes

* Additional libraries
   - FFmpeg filters    : yes
   - FFmpeg devices    : yes
   - inotify           : no (requires inotify)
   - irc               : no (requires irc-client-unix)
   - jemalloc          : no (requires jemalloc)
   - lastfm            : no (requires lastfm)
   - lo                : no (requires lo)
   - memtrace          : no (requires memtrace)
   - osc               : no (requires osc-unix)
   - ssl               : yes
   - sqlite3           : no (requires sqlite3)
   - tls               : yes
   - posix-time2       : no (requires posix)
   - windows service   : no (requires winsvc)
   - YAML support      : no (requires yaml)
   - XML playlists     : no (requires xmlplaylist)

 * Monitoring
   - Prometheus        : yes
```

### Installation method

From OPAM

**Comment by toots** [comment]
Thanks for this report!

Could you provide more specific reproduction steps? I wasn't able to reproduce either using telnet for via the interactive webpage after adding `interactive.harbor()`

**Comment by gAlleb** [comment]
Running `rolling-release-v2.3.x-34-g63aea1b+dev` had time to test too.
Can confirm that everything works fine with this:

```ruby
interactive.harbor()
settings.server.telnet := true
settings.server.telnet.port := 1234

s=sine(440.)

bool = interactive.bool("toggle", false)

thread.when(predicate.changes(bool), {print("bool changed: #{bool()}")})

output(fallible=true, s)

```

Via `http://localhost:8000/interactive` and telnet (`var.set toggle = true`)

**Comment by Russsgithub** [comment]
@toots , @gAlleb , I wrote the script from a much larger script to display behaviour, but didn't test. This test works fine for me too.

I need to go through the script its happening and workout whats going on.

**Comment by Russsgithub** [comment]
When I get the error the variable does update so i guess the problem may be with my script.
.... I'll check the script next time. Sorry.

**Comment by myeungdev** [comment]
I encountered the same issue and found that the functionality itself is working as expected. However, although the interactive value updates successfully, the following error is still appearing in the logs:

```
  [interactive.harbor:3] Could not update variable  (not_found: no default value for list.assoc).
```

After looking into it, I found that the issue is related to how the request body is processed. When you use the interactive web interface to update a value, it sends a payload like this: `⁠var1=false&var2=true&`. The trailing `⁠&` at the end is what causes the error. Since there’s no key-value pair following the ⁠`&`, the ⁠`interactive.liq` expects one to be there, leading to the error. This is always why there is no variable name displayed in the error message.
